pkg/ciliumenvoyconfig/cell.go

```diff
@@ -87,6 +92,13 @@ func registerCECK8sReconciler(params reconcilerParams) {
 		},
 	})
 
+	reconciler.registerResourceWithSyncFn(context.Background(), k8sAPIGroupCiliumEnvoyConfigV2, func() bool {
```

Comment:
Is it guaranteed that this is called before `InitK8sSubsystem`?

Reply:
Yes, as all the constructors are called before any of the start hooks are run (if any constructor fails, the hive is not run at all and no start hooks are run). Changed network policy watcher to do the same.